com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -104,29 +116,160 @@ public int AddEntry(ulong networkObjectId, int behaviourIndex, int variableIndex
             return pos;
         }
 
+        /// <summary>
+        /// Write an Entry to send
+        /// Must match ReadEntry
+        /// </summary>
+        /// <param name="writer">The writer to write the entry to</param>
+        internal void WriteEntry(NetworkWriter writer, in Entry entry)
+        {
+            writer.WriteUInt64(entry.Key.NetworkObjectId);
```

Comment:
I think I have given you my opinnion already on this. NetworkWriter should not be used for just storing something into memory. This should just take a pointer of our current buffer position and then assign `entry` to it and advance the pointer by `sizeof(Entry).

Reply:
That is a good point, and I now agree. What I wonder is on the order of operations. Should we set aside this PR and the next one to fix memory copies first ?

If not, do we want to keep code in PR until all the parts are done ? It's doable, but I fear we'd end up with a big merge and tons of conflict in September. That's why I was shooting for iterative steps.

Reply:
I'm fine with merging this. I just want to make sure that we understand that this is WIP work and that we are aware of another better solution. Maybe just adding a short todo comment like on other comments would help.